### ISSUE TITLE
[ic-stepper] Disabled stepper text color fails contrast ratio checks

### ISSUE THREAD

**Issue by ad9242** [mi6/ic-ui-kit#3422]
## Summary of the bug 
After changes made to bring tokens up to date under #3363, there is still an issue with the disabled step color. This now uses architectural-300 under the covers but this does still not meet required contrast ratios.

This can be recreated by running the ic-stepper visual regression tests. Look for instances of 
// cy.checkA11yWithWait(); A11y failure for disabled text

and uncomment them & run the tests. You can then see that the ic-typography element for the step title has the failure

## 📸 Screenshots or code 

![Image](https://github.com/user-attachments/assets/23cdfa1a-02ca-456a-97c1-a205084ee07c)

**Comment by MI6-255** [comment]
Look into if we need guidance to explain that disabled states do not need to meet this requirement

**Comment by GCHQ-Developer-299** [comment]
After looking into it, I think the solution is to add `aria-hidden` (or `disabled`) to the elements in Disabled step's 'heading-area' 

Looking at an Accessibility FastPass we can see that the Step number doesn't fail accessibility checks, because it already has `aria-hidden` on it. Screenshot below:
<img width="276" alt="Image" src="https://github.com/user-attachments/assets/1cc20946-ec94-4b9f-a341-61aa4a3d527b" />

1. Add `aria-hidden` or, if appropriate, `disabled` to the items in the header section of Disabled steppers, so they are exempt from accessibility checks. 
2. Re-enable the checkA11y line in the tests, it should now pass.
